src/unittest/python/test_thresholding_manager.py

```diff
@@ -0,0 +1,144 @@
+"""
+----------------------------------------------------------------------
+Authors: Stephan Nell
+----------------------------------------------------------------------
+Unit tests for the Thresholding Manager
+----------------------------------------------------------------------
+"""
+import pytest
+import numpy as np
+from image_preprocessing.thresholding_manager import ThresholdingManager
+
+# blank image to test with Height set 1 Width set at 1
+blank_image = np.zeros((1, 1), dtype=np.uint8)
+
+
+def test_thresholding_constructor_incorrect_type():
+    """
+    Test to see if constructor checks for invalid type i.e not String type
+    """
+    with pytest.raises(TypeError):
+        ThresholdingManager(123)
+
+
+def test_thresholding_constructor_paramter_number():
+    """
+    Tests with incorrect number of parameters
+    """
+    with pytest.raises(TypeError):
+        ThresholdingManager("adaptive", 1)
+
+
+def test_thresholding_type():
+    """
+    Tests the verify function with a invalid Thresholding type
+    """
+    with pytest.raises(NameError):
+        ThresholdingManager("Incorrect Thresholding type")
+
+
+def test_thresholding_type_2():
+    """
+    Tests for missing parameter
+    """
+    with pytest.raises(TypeError):
+        ThresholdingManager()
+
+
+def test_apply():
+    """
+    Tests for apply function with Adaptive type
+    """
+    manager = ThresholdingManager("adaptive")
+    manager.apply(blank_image)
```

Comment:
Does `apply` actually change the image that is passed through (1) or does it return the result (2) of applying the threshold?

In both cases, check if the image after `apply` (case 1) or the result of `apply` (case 2) is different from the original.